src/hist1d.jl

```diff
@@ -157,9 +157,26 @@ array. Weight for each value is assumed to be 1.
 """
 function Hist1D(A, r::AbstractRange; overflow=_default_overflow)
     h = Hist1D(Int; bins=r, overflow=overflow)
-    for x in A
-        push!(h, x)
+    firstr = first(r)
+    invstep = inv(step(r))
+    L = nbins(h)
```

Comment:
these are just inlining:

https://github.com/Moelf/FHist.jl/blob/ae5b309d726e5be2a4d246a90b3ad20113c98dc1/src/hist1d.jl#L14-L18

I think we could have consolidated something here... but I need to do `Hist2D` to see the pattern maybe